src/gazebo_parachute_plugin.cpp

```diff
@@ -0,0 +1,131 @@
+
+#include "gazebo_parachute_plugin.h"
+
+namespace gazebo {
+GZ_REGISTER_MODEL_PLUGIN(ParachutePlugin)
+
+ParachutePlugin::ParachutePlugin() : ModelPlugin()
+{
+}
+
+ParachutePlugin::~ParachutePlugin()
+{
+  _updateConnection->~Connection();
+}
+
+void ParachutePlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+{
+  model_ = _model;
+  world_ = model_->GetWorld();
+
+  namespace_.clear();
+  if (_sdf->HasElement("robotNamespace")) {
+    namespace_ = _sdf->GetElement("robotNamespace")->Get<std::string>();
+  } else {
+    gzerr << "[gazebo_parachute_plugin] Please specify a robotNamespace.\n";
+  }
+
+  if (_sdf->HasElement("motorNumber"))
+    motor_number_ = _sdf->GetElement("motorNumber")->Get<int>();
+  else
+    gzerr << "[gazebo_motor_model] Please specify a motorNumber.\n";
+
+  getSdfParam<std::string>(_sdf, "commandSubTopic", trigger_sub_topic_, trigger_sub_topic_);
+
+  // Listen to the update event. This event is broadcast every simulation iteration.
+  _updateConnection = event::Events::ConnectWorldUpdateBegin(boost::bind(&ParachutePlugin::OnUpdate, this, _1));
+
+  node_handle_ = transport::NodePtr(new transport::Node());
+  node_handle_->Init(namespace_);
+
+  trigger_sub_ = node_handle_->Subscribe("~/" + model_->GetName() + trigger_sub_topic_, &ParachutePlugin::VelocityCallback, this);
+}
+
+void ParachutePlugin::OnUpdate(const common::UpdateInfo&){
+  #if GAZEBO_MAJOR_VERSION >= 9
+    physics::ModelPtr parachute_model = world_->ModelByName("parachute_small");
+  #else
+    physics::ModelPtr parachute_model = world_->GetModel("parachute_small");
+  #endif
+    //Trigger parachute if flight termination
+    if(ref_motor_rot_vel_ <= terminate_rot_vel_ ) LoadParachute();
+
+    if(!attached_parachute_ && parachute_model){
+      AttachParachute(parachute_model); //Attach parachute to model
+      attached_parachute_ = true;
+    }
+}
+
+void ParachutePlugin::VelocityCallback(CommandMotorSpeedPtr &rot_velocities) {
+  if(rot_velocities->motor_speed_size() < motor_number_) {
+    std::cout  << "You tried to access index " << motor_number_
+      << " of the MotorSpeed message array which is of size " << rot_velocities->motor_speed_size() << "." << std::endl;
+  } else ref_motor_rot_vel_ = std::min(static_cast<double>(rot_velocities->motor_speed(motor_number_)), static_cast<double>(max_rot_velocity_));
+}
+
+void ParachutePlugin::LoadParachute(){
+  // Don't create duplicate paracutes
+  #if GAZEBO_MAJOR_VERSION >= 9
+    if(physics::ModelPtr parachute_model = world_->ModelByName("parachute_small")) return;
+  #else
+    if(physics::ModelPtr parachute_model = world_->GetModel("parachute_small")) return;
+  #endif
+  // Insert parachute model
+  world_->InsertModelFile("model://parachute_small");
+
+  msgs::Int request;
+  request.set_data(0);
+  
+}
+
+void ParachutePlugin::AttachParachute(physics::ModelPtr &parachute_model){
+
+#if GAZEBO_MAJOR_VERSION >= 9
+  const ignition::math::Pose3d uavPose = model_->WorldPose();
```

Comment:
Can we change the naming from `uavPose` to `vehicle_pose`? Also, the naming format doesn't seem to be uniform (some camel case, then some snake case). Let's keep the variables snake case, unless you have another suggestion.

Reply:
Done!